streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -142,6 +146,25 @@ public StreamTask(final TaskId id,
     }
 
     /**
+     * Finds the source of the given topic partition.
+     *
+     * @param partition the topic partition
+     * @return
+     */
+    private SourceNode findSource(final TopicPartition partition) {
```

Comment:
You are right. The main reason that I put this extra function is that I was getting many fails in tests. 

There were some tests with no source nodes defined in the test topology  or the defined source nodes are not related with partitions.